File: src/main/java/com/example/entity/Url.java
package com.example.entity;

import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.sql.Timestamp;

@Entity(name = "URL")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Url {

    @Id
    @GeneratedValue
    @Column(name = "id", columnDefinition = "serial")
    private Integer id;

    @Column(nullable = false)
    private String key;

    @Column(nullable = false, name = "short_url")
    private String shortUrl;

    @Column(nullable = false, name = "long_url")
    private String longUrl;

    @Column(nullable = false, name = "created_at")
    private Timestamp createdAt;

    @Column(nullable = false, name = "updated_at")
    private Timestamp updatedAt;
}

File: src/main/java/com/example/entity/Url.kt
package com.example.entity

import io.quarkus.hibernate.orm.panache.kotlin.PanacheEntity
import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import java.sql.Timestamp
import java.time.LocalDateTime

@Entity
class Url : PanacheEntity() {

    @Id
    @GeneratedValue
    override var id: Long? = null;
    lateinit var shortUrl: String
    lateinit var longUrl: String
    lateinit var createdAt: Timestamp
    lateinit var updatedAt: Timestamp

}